browser_utils.js

```diff
@@ -557,6 +557,34 @@ async function clickSelector(page, selector, {timeout=getDefaultTimeout(page), c
     }
 }
 
+/**
+ * Asserts that a selector is not present in the passed page or frame.
+ *
+ * @example
+ * ```javascript
+ * await assertNotSelector(page, 'div[data-id="foo"] a.view', {message: 'Expected foo to not be present'});
+ * ```
+ * @param {import('puppeteer').Page} page puppeteer page object.
+ * @param {string} selector [CSS selector](https://www.w3.org/TR/2018/REC-selectors-3-20181106/#selectors) (aka query selector) of the targeted element.
+ * @param {{timeout?: number, message?: string}} [__namedParameters] Options (currently not visible in output due to typedoc bug)
+ * @param {string?} message Error message shown if the element is not visible in time.
+ * @param {number?} timeout How long to wait, in milliseconds.
+ */
+async function assertNotSelector(page, selector, {timeout=getDefaultTimeout(page), message} = {}) {
+    const config = getBrowser(page)._pentf_config;
+    addBreadcrumb(config, `enter assertNotSelector(${selector})`);
+    try {
+        await page.waitForSelector(selector, {timeout});
```

Comment:
There is a potential flakiness problem here: If the page quickly flashes the testId (or the testId sometimes is present on start) and then it goes away, the test will fail. That's why `assertNotXPath` has a different implementation.